lib/artemis/client.rb

```diff
@@ -303,15 +348,20 @@ def initialize(connection, callbacks, default_context) #:nodoc:
       end
 
       def execute(document:, operation_name: nil, variables: {}, context: {}) #:nodoc:
-        _context = @default_context.deep_merge(context)
+        context = @default_context.deep_merge(context)
```

Comment:
Here it is important to assign a new variable to a local variable to avoid mutating the original `context` by accident.

Reply:
Not sure how that would happen but using `_` infront of a variable means it's unused. Maybe a different name is better then `_context`?